src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -521,189 +518,159 @@ mbedtls_ecp_group_id MapECPGroupId(SupportedECPKeyTypes keyType)
-
-    VerifyOrExit(result == 0, error = CHIP_ERROR_INTERNAL);
+    CHIP_ERROR error                = CHIP_NO_ERROR;
+    psa_status_t status             = PSA_ERROR_BAD_STATE;
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    mbedtls_svc_key_id_t keyId      = 0;
+    size_t outputLen                = 0;
+    const psa_ecp_keypair * keypair = to_const_keypair(&mKeypair);
 
-    VerifyOrExit((mbedtls_mpi_size(&r) <= kP256_FE_Length) && (mbedtls_mpi_size(&s) <= kP256_FE_Length),
-                 error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(mInitialized, error = CHIP_ERROR_WELL_UNINITIALIZED);
+    VerifyOrExit((msg != nullptr) && (msg_length > 0), error = CHIP_ERROR_INVALID_ARGUMENT);
 
-    // Concatenate r and s to output. Sizes were checked above.
-    result = mbedtls_mpi_write_binary(&r, out_signature.Bytes() + 0u, kP256_FE_Length);
-    VerifyOrExit(result == 0, error = CHIP_ERROR_INTERNAL);
+    psa_set_key_type(&attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+    psa_set_key_bits(&attributes, keypair->mBitLength);
+    psa_set_key_algorithm(&attributes, PSA_ALG_ECDSA(PSA_ALG_SHA_256));
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_SIGN_MESSAGE);
 
-    result = mbedtls_mpi_write_binary(&s, out_signature.Bytes() + kP256_FE_Length, kP256_FE_Length);
-    VerifyOrExit(result == 0, error = CHIP_ERROR_INTERNAL);
+    status = psa_import_key(&attributes, keypair->mPrivateKey, PSA_BITS_TO_BYTES(keypair->mBitLength), &keyId);
+    VerifyOrExit(status == PSA_SUCCESS, error = CHIP_ERROR_INTERNAL);
 
-    VerifyOrExit(out_signature.SetLength(kP256_ECDSA_Signature_Length_Raw) == CHIP_NO_ERROR, error = CHIP_ERROR_INTERNAL);
+    status = psa_sign_message(keyId, PSA_ALG_ECDSA(PSA_ALG_SHA_256), msg, msg_length, out_signature.Bytes(),
+                              out_signature.Capacity(), &outputLen);
+    VerifyOrExit(status == PSA_SUCCESS, error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(outputLen == kP256_ECDSA_Signature_Length_Raw, error = CHIP_ERROR_INTERNAL);
+    SuccessOrExit(out_signature.SetLength(outputLen) == CHIP_NO_ERROR);
 
 exit:
-    keypair = nullptr;
-    mbedtls_ecdsa_free(&ecdsa_ctxt);
-    mbedtls_mpi_free(&s);
-    mbedtls_mpi_free(&r);
-    logMbedTLSError(result);
+    logPsaError(status);
+    if (0 != keyId)
+    {
+        psa_destroy_key(keyId);
+    }
+    psa_reset_key_attributes(&attributes);
     return error;
-#else
-    return CHIP_ERROR_NOT_IMPLEMENTED;
-#endif
 }
 
 CHIP_ERROR P256PublicKey::ECDSA_validate_msg_signature(const uint8_t * msg, const size_t msg_length,
                                                        const P256ECDSASignature & signature) const
 {
-#if defined(MBEDTLS_ECDSA_C)
-    VerifyOrReturnError((msg != nullptr) && (msg_length > 0), CHIP_ERROR_INVALID_ARGUMENT);
+    CHIP_ERROR error                = CHIP_NO_ERROR;
+    psa_status_t status             = PSA_ERROR_BAD_STATE;
```

Comment:
I think we don't need to initialize this variable, as it is never used before assignment of method's output result. Similarly in other places.